experiments/tf_trainer/keras_gru_attention/run.py

```diff
@@ -35,36 +30,52 @@
 }  # type: Dict[str, tf.DType]
 
 
+class KerasGRUAttentionModelRunner(model_runner.ModelRunner):
+
+  def __init__(self, embeddings_path: str, text_feature: str,
+               labels: Dict[str, tf.DType],
+               text_preprocessor: text_preprocessor.TextPreprocessor) -> None:
+    self._embeddings_path = embeddings_path
+    self._text_feature = text_feature
+    self._labels = labels
+    self._text_preprocessor = text_preprocessor
+    nltk.download("punkt")
+
+  def dataset_input(self, train_path, validate_path):
+    return tfrecord_input.TFRecordInput(
+        train_path=train_path,
+        validate_path=validate_path,
+        text_feature=self._text_feature,
+        labels=self._labels,
+        feature_preprocessor=self._text_preprocessor.tokenize_tensor_op(
+            nltk.word_tokenize))
+
+  def estimator(self, model_dir):
+    estimator_no_embedding = model.KerasRNNModel(set(
+        self._labels.keys())).get_estimator(model_dir)
+
+    # TODO: Move embedding into Keras model.
+    estimator = self._text_preprocessor.create_estimator_with_embedding(
+        estimator_no_embedding, self._text_feature)
+
+    return estimator
+
+
 def main(argv):
   del argv  # unused
 
-  nltk.download("punkt")
-
-  train_path = types.Path(FLAGS.train_path)
-  validate_path = types.Path(FLAGS.validate_path)
-  embeddings_path = types.Path(FLAGS.embeddings_path)
+  embeddings_path = FLAGS.embeddings_path
   text_feature_name = FLAGS.text_feature_name
-  model_dir = FLAGS.model_dir
 
   preprocessor = text_preprocessor.TextPreprocessor(embeddings_path)
-  dataset = tfrecord_input.TFRecordInput(
-      train_path=train_path,
-      validate_path=validate_path,
+
+  runner = KerasGRUAttentionModelRunner(
+      embeddings_path=embeddings_path,
       text_feature=text_feature_name,
       labels=LABELS,
-      feature_preprocessor=preprocessor.tokenize_tensor_op(nltk.word_tokenize))
-
-  estimator_no_embedding = model.KerasRNNModel(set(
-      LABELS.keys())).get_estimator(model_dir)
-
-  # TODO: Move embedding into Keras model.
-  estimator = preprocessor.create_estimator_with_embedding(
-      estimator_no_embedding, text_feature_name)
+      text_preprocessor=preprocessor)
 
-  for _ in range(20):
-    estimator.train(input_fn=dataset.train_input_fn, steps=1000)
-    metrics = estimator.evaluate(input_fn=dataset.validate_input_fn, steps=100)
-    tf.logging.info(metrics)
+  runner.train_with_eval(20000, 1000, 100)
```

Comment:
Because these are serving as templates for future model building, I'd like to make it more obvious what the values that are being given to this function are. Maybe you can make flags (in model runner) with these default values so that everyone knows that these correspond to steps, eval_period, eval_steps.

Reply:
ditto